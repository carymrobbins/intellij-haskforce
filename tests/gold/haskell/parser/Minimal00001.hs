module Minimal00001 where

class HasFoo a where
    {-# MINIMAL foo #-}
    foo :: a -> String
    bar :: a -> String

class HasFoo' a where
    foo' :: a -> String
    {-# MINIMAL foo' #-}
    bar' :: a -> String

class HasFoo'' a where
    foo'' :: a -> String
    bar'' :: a -> String
    {-# MINIMAL foo'' #-}
