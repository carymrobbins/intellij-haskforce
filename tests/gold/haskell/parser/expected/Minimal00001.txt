Haskell File
  MODULEDECL
    PsiElement(HaskellTokenType.module)('module')
    QCONID
      CONID
        PsiElement(HaskellTokenType.conidRegexp)('Minimal00001')
    PsiElement(HaskellTokenType.where)('where')
  BODY
    CLASSDECL
      PsiElement(HaskellTokenType.class)('class')
      CTYPE
        TYPEE
          ATYPE
            OQTYCON
              QTYCON
                TYCON
                  CONID
                    PsiElement(HaskellTokenType.conidRegexp)('HasFoo')
          ATYPE
            TYVAR
              VARID
                PsiElement(HaskellTokenType.varidRegexp)('a')
      PsiElement(HaskellTokenType.where)('where')
      PPRAGMA
        PsiElement(HaskellTokenType.{-#)('{-#')
        PsiElement(HaskellTokenType.PRAGMA)('MINIMAL')
        PsiElement(HaskellTokenType.PRAGMA)('foo')
        PsiElement(HaskellTokenType.#-})('#-}')
      CDECL
        GENDECL
          VARS
            VARID
              PsiElement(HaskellTokenType.varidRegexp)('foo')
          PsiElement(HaskellTokenType.::)('::')
          CTYPE
            TYPEE
              ATYPE
                TYVAR
                  VARID
                    PsiElement(HaskellTokenType.varidRegexp)('a')
              PsiElement(HaskellTokenType.->)('->')
              TYPEE
                ATYPE
                  OQTYCON
                    QTYCON
                      TYCON
                        CONID
                          PsiElement(HaskellTokenType.conidRegexp)('String')
      CDECL
        GENDECL
          VARS
            VARID
              PsiElement(HaskellTokenType.varidRegexp)('bar')
          PsiElement(HaskellTokenType.::)('::')
          CTYPE
            TYPEE
              ATYPE
                TYVAR
                  VARID
                    PsiElement(HaskellTokenType.varidRegexp)('a')
              PsiElement(HaskellTokenType.->)('->')
              TYPEE
                ATYPE
                  OQTYCON
                    QTYCON
                      TYCON
                        CONID
                          PsiElement(HaskellTokenType.conidRegexp)('String')
    CLASSDECL
      PsiElement(HaskellTokenType.class)('class')
      CTYPE
        TYPEE
          ATYPE
            OQTYCON
              QTYCON
                TYCON
                  CONID
                    PsiElement(HaskellTokenType.conidRegexp)('HasFoo'')
          ATYPE
            TYVAR
              VARID
                PsiElement(HaskellTokenType.varidRegexp)('a')
      PsiElement(HaskellTokenType.where)('where')
      CDECL
        GENDECL
          VARS
            VARID
              PsiElement(HaskellTokenType.varidRegexp)('foo'')
          PsiElement(HaskellTokenType.::)('::')
          CTYPE
            TYPEE
              ATYPE
                TYVAR
                  VARID
                    PsiElement(HaskellTokenType.varidRegexp)('a')
              PsiElement(HaskellTokenType.->)('->')
              TYPEE
                ATYPE
                  OQTYCON
                    QTYCON
                      TYCON
                        CONID
                          PsiElement(HaskellTokenType.conidRegexp)('String')
    PPRAGMA
      PsiElement(HaskellTokenType.{-#)('{-#')
      PsiElement(HaskellTokenType.PRAGMA)('MINIMAL')
      PsiElement(HaskellTokenType.PRAGMA)('foo'')
      PsiElement(HaskellTokenType.#-})('#-}')
    GENDECL
      VARS
        VARID
          PsiElement(HaskellTokenType.varidRegexp)('bar'')
      PsiElement(HaskellTokenType.::)('::')
      CTYPE
        TYPEE
          ATYPE
            TYVAR
              VARID
                PsiElement(HaskellTokenType.varidRegexp)('a')
          PsiElement(HaskellTokenType.->)('->')
          TYPEE
            ATYPE
              OQTYCON
                QTYCON
                  TYCON
                    CONID
                      PsiElement(HaskellTokenType.conidRegexp)('String')
    CLASSDECL
      PsiElement(HaskellTokenType.class)('class')
      CTYPE
        TYPEE
          ATYPE
            OQTYCON
              QTYCON
                TYCON
                  CONID
                    PsiElement(HaskellTokenType.conidRegexp)('HasFoo''')
          ATYPE
            TYVAR
              VARID
                PsiElement(HaskellTokenType.varidRegexp)('a')
      PsiElement(HaskellTokenType.where)('where')
      CDECL
        GENDECL
          VARS
            VARID
              PsiElement(HaskellTokenType.varidRegexp)('foo''')
          PsiElement(HaskellTokenType.::)('::')
          CTYPE
            TYPEE
              ATYPE
                TYVAR
                  VARID
                    PsiElement(HaskellTokenType.varidRegexp)('a')
              PsiElement(HaskellTokenType.->)('->')
              TYPEE
                ATYPE
                  OQTYCON
                    QTYCON
                      TYCON
                        CONID
                          PsiElement(HaskellTokenType.conidRegexp)('String')
      CDECL
        GENDECL
          VARS
            VARID
              PsiElement(HaskellTokenType.varidRegexp)('bar''')
          PsiElement(HaskellTokenType.::)('::')
          CTYPE
            TYPEE
              ATYPE
                TYVAR
                  VARID
                    PsiElement(HaskellTokenType.varidRegexp)('a')
              PsiElement(HaskellTokenType.->)('->')
              TYPEE
                ATYPE
                  OQTYCON
                    QTYCON
                      TYCON
                        CONID
                          PsiElement(HaskellTokenType.conidRegexp)('String')
    PPRAGMA
      PsiElement(HaskellTokenType.{-#)('{-#')
      PsiElement(HaskellTokenType.PRAGMA)('MINIMAL')
      PsiElement(HaskellTokenType.PRAGMA)('foo''')
      PsiElement(HaskellTokenType.#-})('#-}')